### PR TITLE
Updated EBS code example docs

### PR DESCRIPTION
**Pull request by Doug-AWS**
*Issue #, if available:*
We had one EBS code example, `create-snapshot`, which had no docs.

*Description of changes:*
- Added doc comments.
- Added `delete-snapshot`, `get-snapshot-state`, and `list-snapshots` code examples.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/sdk/examples/ebs/src/bin/create-snapshot.rs [hunk 1]
```diff
@@ -52,27 +102,15 @@ async fn main() -> Result<(), aws_sdk_ebs::Error> {
...
+    client
         .complete_snapshot()
         .changed_blocks_count(2)
         .snapshot_id(&snapshot_id)
         .send()
         .await?;
-    println!("snapshot complete: {:#?}", rsp);
-
-    // NOTE: you need to wait for `status != pending`
-    let ec2_client = aws_sdk_ec2::Client::from_env();
```
Comment (rcoh) [inline aws/sdk/examples/ebs/src/bin/create-snapshot.rs:64]: can you add a reference to your `get-snapshot-state` example?

File: aws/sdk/examples/ebs/src/bin/get-snapshot-state.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
+/*
```
Comment (rcoh) [inline aws/sdk/examples/ebs/src/bin/get-snapshot-state.rs:1]: can you add a comment calling out that this actually uses the ec2 API?

File: aws/sdk/examples/ebs/src/bin/list-snapshots.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+        println!();
+    }
+
+    let config = Config::builder().region(region_provider).build();
+    let client = Client::from_conf(config);
+
+    // You can find snapshots for any account by replacing
+    // "self" with their account ID.
+    let resp = client.describe_snapshots().owner_ids("self").send().await?;
```
Comment (rcoh) [inline aws/sdk/examples/ebs/src/bin/list-snapshots.rs:55]: is `self` a special account id?

**Review by rcoh — CHANGES_REQUESTED** [review]

File: aws/sdk/examples/ebs/src/bin/create-snapshot.rs (hunk 1 above)
Comment (Doug-AWS) [reply aws/sdk/examples/ebs/src/bin/create-snapshot.rs:64]: Will do.

File: aws/sdk/examples/ebs/src/bin/list-snapshots.rs (hunk 3 above)
Comment (Doug-AWS) [reply aws/sdk/examples/ebs/src/bin/list-snapshots.rs:55]: Yes, it means whoever is running the code's account ID. I'll add that explanation as a comment.

File: aws/sdk/examples/ebs/src/bin/get-snapshot-state.rs (hunk 2 above)
Comment (Doug-AWS) [reply aws/sdk/examples/ebs/src/bin/get-snapshot-state.rs:1]: Done.

**Review by rcoh — APPROVED** [review]

**Pull request merged**